### ISSUE TITLE
Many - to - Many loading correctly, but not initializing

### ISSUE THREAD

**Issue by sydhenry** [vuex-orm/vuex-orm#142]
I am currently following the [Defining Relationships](https://vuex-orm.github.io/vuex-orm/relationships/defining-relationships.html) documentation and can't get the relations to be retrieved.

Using Vue.js devtools you can see when it is on my component, it doesn't have it's tags.
![screenshot_20180411_143033](https://user-images.githubusercontent.com/12238856/38645304-780939d0-3d98-11e8-84bc-7a7bbfcc497a.png)

To be sure it wasn't my `components/Event.vue` erasing it, on a blank page I called the events.
`$vm0.$store.dispatch('entities/event/refreshEvents')`

I check with dev tools and the tag is in Vuex
![screenshot_20180411_143103](https://user-images.githubusercontent.com/12238856/38645645-a627aa44-3d99-11e8-9e83-125af8ff6b72.png)

I then call for the specific event using withAll()
`$vm0.$store.getters['entities/event/query']().withAll().find("5ac4028b930c5c4c441fe925")`

But that produces the event, but without the tags:

![screenshot_20180411_150126](https://user-images.githubusercontent.com/12238856/38645564-5b2edfd0-3d99-11e8-9ce0-81e98293437d.png)

I'm not sure what I am doing wrong here. It is odd because the tag that is listed isn't created as a tag manually, but creates it through the relationship. I have the models that I am using below.

``` js
// Event.js Model
//

import { Model } from '@vuex-orm/core'
import Tag from './Tag'
import Taggable from './Taggable'
import moment, { moment2string, string2moment } from '../../../plugins/moment'

export default class extends Model {
  // This is the name used as module name of the Vuex Store.
  static entity = 'event'

  // List of all fields (schema) of the post model. `this.attr` is used
  // for the generic field type. The argument is the default value.

  static fields () {
    return {
      id: this.string(''),
      title: this.string(''),
      start: this.attr(''),
      end: this.attr(''),
      notes: this.string(''),
      tags: this.morphToMany(Tag, Taggable, 'tagId', 'taggableId', 'taggableType'),
      // tags: this.belongsToMany(Tag, Taggable, 'eventId', 'tagId'),

      age: this.number(0),
      base_age: this.number(0),
      modified: this.attr(''),
      created: this.attr(''),
      edits: this.number(0)
    }
  }

  static mutators () {
    return {
      start (value) {
        return string2moment(value)
      },
      end (value) {
        return string2moment(value)
      },
      created (value) {
        return string2moment(value)
      },
      modified (value) {
        return string2moment(value)
      }
    }
  }

  /**
   * Convert moment object to string for storage/ api calls
   */
  datetimeString (datetime) {
    return moment2string(datetime)
  }

  /**
   * Diff shortcut for moment.js
   * @returns number
   */
  get diff () {
    return this.start.diff(this.end)
  }
}

// Tag.js Model
//

import { Model } from '@vuex-orm/core'
// import Event from './Event'
// import Taggable from './Taggable'

export default class extends Model {
  // This is the name used as module name of the Vuex Store.
  static entity = 'tag'

  // List of all fields (schema) of the post model. `this.attr` is used
  // for the generic field type. The argument is the default value.
  static fields () {
    return {
      id: this.string(''),
      text: this.string(''),
      description: this.string('')
      // either of these cause
      // TypeError: related is undefined
      // events: this.morphedByMany(Event, Taggable, 'tagId', 'taggableId', 'taggableType')
      // alias: this.morphedByMany(this, Taggable, 'tagId', 'taggableId', 'taggableType')
    }
  }
}


// Taggable.js Model
//

import { Model } from '@vuex-orm/core'

export default class extends Model {
  // This is the name used as module name of the Vuex Store.
  static entity = 'taggable'

  static primaryKey = ['tagId', 'taggableId']
  // List of all fields (schema) of the post model. `this.attr` is used
  // for the generic field type. The argument is the default value.
  static fields () {
    return {
      tagId: this.attr(null),
      eventId: this.attr(null),
      taggableId: this.attr(null),
      taggableType: this.attr(null),
      taggable: this.morphTo('taggableId', 'taggableType')
    }
  }
}
```

**Comment by kiaking** [comment]
Hi! Thank you for the report.

The reason why you get empty array in `tags` in `events` when fetching via getter is because getter will remove any relationship if there isn't one. So that is expected, but if you do have relationship, it should fetch the object. 

I've tested at my side and it did reproduce. It fixed by using plural on `entity` name. Could you try changing `static entity = 'tag'` to be `static entity = 'tags'` and see if it fixes?

I will look into more on why the hell this is happening.

**Comment by kiaking** [comment]
Daah I found a fix. I'll fix this soon. Could you workaround this issue by making the entity name of Tag to match the key that defines the morphToMany? In your example the entity name of the Tag should be `tags`.

**Comment by sydhenry** [comment]
So I made that change and while I haven't gotten it to work in my component (it has been what 5 min?) but `$vm0.$store.getters['entities/event/query']().withAll().find("5ac4028b930c5c4c441fe925")` does return the tags Array.

This fell under the 'no way that is the issue' category for me. I was just about to go to sleep, but the plurality of the `entity` is important, and fixes the issue as far as I can tell. I'll get back to you tomorrow with more information.

**Comment by kiaking** [comment]
Yes, sorry for that. It's actually not the plural issue, currently Event is trying to look for related entity name when creating pivot table which is `Tag.entity === 'tag'` where it actually has to find the key name for the relationship field which is `tags`. So if you change `Tag.entity` to be `tags`, it matches the key name so it succeed on retrieving... Really stupid issue but test case was using same names so it was passing 😢

**Comment by kiaking** [comment]
This has been resolved by Vuex ORM 0.24.4! Sorry for the inconvenience!
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.24.4
